pdfminer/pdfinterp.py

```diff
@@ -331,11 +332,22 @@ def do_keyword(self, pos: int, token: PSKeyword) -> None:
                 if len(objs) % 2 != 0:
                     error_msg = f"Invalid dictionary construct: {objs!r}"
                     raise PSTypeError(error_msg)
-                d = {literal_name(k): v for (k, v) in choplist(2, objs)}
-                (pos, data) = self.get_inline_data(pos + len(b"ID "))
+                d = {literal_name(k): resolve1(v) for (k, v) in choplist(2, objs)}
+                eos = b"EI"
+                filter = d.get("F", None)
+                if filter is not None:
+                    if isinstance(filter, PSLiteral):
+                        filter = [filter]
+                    for f in LITERALS_ASCII85_DECODE:
+                        if f in filter:
+                            eos = b"~>"
```

Comment:
I'm wondering if this is correct if ASCII85 is not the first filter. In other words, has a FlateDecode ASCII85 stream the same end-of-stream marker as an ASCII85 FlateDecode stream. I suspect the first filter determines the end-of-stream marker.

Reply:
Ah, good catch.  Exactly, it should only apply if ASCII85 is the first filter (in the case of this specific document it is followed by FlateDecode).  I'll update in a second!